internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_providers.go

```diff
@@ -227,15 +255,27 @@ func dataSourceMongoDBAtlasFederatedSettingsIdentityProvidersRead(ctx context.Co
 	if !federationSettingsIDOk {
 		return diag.FromErr(errors.New("federation_settings_id must be configured"))
 	}
-	// There is no available paginated method
-	// If no protocol is specified, only SAML identity providers will be returned. Use ListIdentityProvidersWithParams to specify protocol
-	federatedSettingsIdentityProviders, _, err := connV2.FederatedAuthenticationApi.ListIdentityProviders(ctx, federationSettingsID.(string)).Execute()
 
-	if err != nil {
-		return diag.Errorf("error getting federatedSettings IdentityProviders assigned (%s): %s", federationSettingsID, err)
+	oidcParams := &admin.ListIdentityProvidersApiParams{
+		FederationSettingsId: federationSettingsID.(string),
+		Protocol:             conversion.StringPtr("OIDC"),
+	}
+	samlParams := &admin.ListIdentityProvidersApiParams{
+		FederationSettingsId: federationSettingsID.(string),
+		Protocol:             conversion.StringPtr("SAML"),
+	}
```

Comment:
can both protocols be enabled at the same time?

Reply:
One federation can have multiple IdPs, so you can have one SAML IdP and one OIDC IdP in the same federation. But one IdP can be either SAML or OIDC, not both